external-builds/pytorch/build_prod_wheels.py

```diff
@@ -174,7 +174,16 @@ def directory_if_exists(dir: Path) -> Path | None:
 def do_install_rocm(args: argparse.Namespace):
     # Because the rocm-sdk package caches current GPU selection and such, we
     # always purge it to ensure a clean rebuild.
-    exec([sys.executable, "-m", "pip", "cache", "remove", "rocm_sdk"], cwd=Path.cwd())
+    exec(
+        [sys.executable, "-m", "pip", "cache", "remove", "rocm_sdk"],
+        cwd=Path.cwd(),
+        env={
+            **os.environ,
+            "PIP_CACHE_DIR": (
+                str(args.pip_cache_dir) if args.pip_cache_dir else "/tmp/pipcache"
```

Comment:
Let's not do it like this: the `cache remove` command takes an optional `--cache-dir` argument. Add it to the command line vs mucking with the environment, and don't default it to /tmp/pipcache. If no `--cache-dir` to this script, then don't pass a `--cache-dir` to pip commands.

Reply:
You can just create a local variable `cache_dir_args` and leave it as an empty list if not set, otherwise `["--cache-dir", args.pip_cache_dir]`. Then use that here and a dozen lines below.